<launch>

  <!-- args corresponding to environment variables -->
  <arg name="UAV_NAME" default="$(env UAV_NAME)" />
  <arg name="LOGGER_DEBUG" default="$(optenv LOGGER_DEBUG false)" />

    <!-- other args -->
  <arg name="standalone" default="false" />
  <arg name="debug" default="false" />
  <arg name="nodelet_manager_name" default="$(arg UAV_NAME)_mapping_nodelet_manager" />
  <arg name="world_frame_id" default="$(arg UAV_NAME)/world_origin" />

    <!-- custom configs -->
  <arg name="config_pcl_filter_ouster" default="" />
  <arg name="config_octomap_server" default="" />
  <arg name="config_octomap_planner" default="" />
  <arg name="config_octomap_visualizer" default="" />

    <!-- NodeletManager //{ -->

  <include file="$(find mrs_octomap_mapping_planning)/launch/nodelet_manager.launch">

    <arg name="standalone" value="$(arg standalone)" />
    <arg name="debug" value="$(arg debug)" />
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)" />

  </include>

  <!-- //} -->

  <!-- OctomapServer //{ -->

  <include file="$(find mrs_octomap_server)/launch/octomap.launch">

    <arg name="standalone" value="$(arg standalone)" />
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)" />
    <arg name="custom_config" value="$(arg config_octomap_server)" />
    <arg name="debug" value="false" />
    <arg name="lidar_3d_topic_0_in" value="lidar/points" />
    <arg name="lidar_3d_topic_0_over_max_range_in" value="pcl_filter_ouster/points_over_max_range" />
    <arg name="depth_camera_topic_0_in" value="pcl_filter_rs_front/points_processed" />
    <arg name="depth_camera_topic_0_over_max_range_in" default="pcl_filter_rs_front/points_over_max_range" />
    <arg name="depth_camera_topic_1_in" value="pcl_filter_rs_down/points_processed" />
    <arg name="depth_camera_topic_1_over_max_range_in" default="pcl_filter_rs_down/points_over_max_range" />
    <arg name="camera_info_topic_0_in" default="front_rgbd/aligned_depth_to_color/camera_info" />
    <arg name="camera_info_topic_1_in" default="down_rgbd/aligned_depth_to_color/camera_info" />
    <arg name="lidar_2d_topic_0_in" value="rplidar/scan" />
    <arg name="world_frame_id" value="$(arg world_frame_id)" />

  </include>

  <!-- //} -->

  <!-- OctomapVisualizer //{ -->

  <include file="$(find mrs_octomap_tools)/launch/octomap_rviz_visualizer.launch">

    <arg name="standalone" value="$(arg standalone)" />
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)" />
    <arg name="custom_config" value="$(arg config_octomap_visualizer)" />
    <arg name="debug" value="false" />
    <arg name="node_name" value="octomap_global_vis" />
    <arg name="octomap_topic" value="octomap_server/octomap_global_full" />

  </include>

  <include file="$(find mrs_octomap_tools)/launch/octomap_rviz_visualizer.launch">

    <arg name="standalone" value="$(arg standalone)" />
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)" />
    <arg name="custom_config" value="$(arg config_octomap_visualizer)" />
    <arg name="debug" value="false" />
    <arg name="node_name" value="octomap_local_vis" />
    <arg name="octomap_topic" value="octomap_server/octomap_local_full" />

  </include>

  <!-- //} -->

  <!-- OctomapPlanner //{ -->

  <include file="$(find mrs_octomap_planner)/launch/octomap_planner.launch">

    <arg name="standalone" value="$(arg standalone)" />
    <arg name="nodelet_manager_name" value="$(arg nodelet_manager_name)" />
    <arg name="custom_config" value="$(arg config_octomap_planner)" />
    <arg name="debug" value="false" />

  </include>

  <!-- //} -->

</launch>
